bot_action/bot_action.py

```
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+#
+# SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import os
+import requests
+
+
+def get_suggestion(issue_body: str) -> str:
+    payload = json.dumps({'integration_id': os.environ['BOT_INTEGRATION_ID'], 'query': issue_body})
+
+    headers = {'content-type': 'application/json', 'X-API-KEY': os.environ['BOT_API_KEY']}
+    r = requests.post(os.environ['BOT_API_ENDPOINT'], data=payload, headers=headers)
+
+    answer = r.json()['answer']
+    assert isinstance(answer, str), 'No answer found'
```

Feel free to ignore. Would it make sense to post something even if there is no answer. Or maybe just exit gracefully. IOW is no answer reason to fail the action or a valid state?

---

If there is a green passing action then one hardly notices that there is some issue. I don't like these silent failures and usually notice them too late. So I'd like this to fail. It won't block anything. Just it will be more visible in the actions tab that one needs to look at the reason of the failure.

---

Thank you! LGTM